src/Fl_Graphics_Driver.cxx

```diff
@@ -932,6 +940,50 @@ void Fl_Scalable_Graphics_Driver::pie(int x,int y,int w,int h,double a1,double a
   pie_unscaled(xx, yy, w, h, a1, a2);
 }
 
+
+void Fl_Scalable_Graphics_Driver::small_circle(int x0, int y0, int d, Fl_Color c) {
+  Fl_Color saved = color();
+
+#define DEBUG_DRAW_CIRCLE (0) // bit 1 = draw bounding box (green)
+#if (DEBUG_DRAW_CIRCLE & 1)
+  fl_rectf(x0, y0, d, d, FL_GREEN);
+#endif
+  color(c);
```

Comment:
The above debugging code **must** be in the main function `fl_draw_circle()` because it is intended as a debugging aid for widget developers. It was a great help for me when I rewrote widgets to use fl_draw_circle() - not only for small circles.

Therefore it must not be only in the special driver method that uses rectangle drawing but in the main function.

Saving and restoring the color setting should also be done in `fl_draw_circle()` and could be removed here.

Reply:
Committed in 80d102d